events/janus_wsevh.c

```diff
@@ -575,60 +576,55 @@ json_t *janus_wsevh_handle_request(json_t *request) {
 		}
 }
 
-/* Thread to implement the WebSockets loop */
+#if (LWS_LIBRARY_VERSION_MAJOR >= 3 && LWS_LIBRARY_VERSION_MINOR >= 2) || (LWS_LIBRARY_VERSION_MAJOR >= 4)
+/*
+* Websocket thread loop for websocket library newer than 3.2
+* The reconnect is handled in a dedicated lws scheduler janus_wsevh_schedule_connect_attempt
+*/
 static void *janus_wsevh_thread(void *data) {
-	JANUS_LOG(LOG_VERB, "Joining WebSocketsEventHandler client thread\n");
+	JANUS_LOG(LOG_VERB, "Joining WebSocketsEventHandler (lws>=3.2) client thread\n");
+	int n = 0;
+	while(n >= 0 && g_atomic_int_get(&initialized) && !g_atomic_int_get(&stopping))
+		n = lws_service(context, 0);
+	lws_context_destroy(context);
+	JANUS_LOG(LOG_VERB, "Leaving WebSocketsEventHandler (lws>=3.2) client thread\n");
+	return NULL;
+}
+#else
+/*
+* Websocket thread loop for websocket library prior to (less than) 3.2
+* The reconnect is handled in the loop for lws < 3.2
+*/
+static void *janus_wsevh_thread(void *data) {
+	JANUS_LOG(LOG_VERB, "Joining WebSocketsEventHandler (lws<3.2) client thread\n");
 	while(g_atomic_int_get(&initialized) && !g_atomic_int_get(&stopping)) {
 		/* Loop until we have to stop */
 		if(!reconnect) {
 			lws_service(context, 50);
 		} else {
 			/* We should reconnect, get rid of the previous context */
-			if(reconnect_retries > 0) {
+			if(reconnect_delay > 0) {
 				/* Wait a few seconds before retrying */
 				gint64 now = janus_get_monotonic_time();
-				if((now-disconnected) < (gint64)reconnect_retries*G_USEC_PER_SEC) {
+				if((now-disconnected) < (gint64)reconnect_delay*G_USEC_PER_SEC) {
 					/* Try again later */
 					g_usleep(100000);
 					continue;
 				}
 			}
-			if(reconnect_retries == 0)
-				reconnect_retries++;
-			else if (reconnect_retries < JANUS_WSEVH_MAX_RETRY_SECS) {
-				reconnect_retries += reconnect_retries;
-				if (reconnect_retries > JANUS_WSEVH_MAX_RETRY_SECS)
-					reconnect_retries = JANUS_WSEVH_MAX_RETRY_SECS;
-			}
-			JANUS_LOG(LOG_WARN, "Reconnecting to WebSockets event handler backend... (next retry in %ds)\n",
-				reconnect_retries);
 			ws_client = NULL;
-			wsi = NULL;
-			struct lws_client_connect_info i;
-			memset(&i, 0, sizeof(i));
-			i.host = address;
-			i.origin = address;
-			i.address = address;
-			i.port = port;
-			i.path = path;
-			i.context = context;
-			i.ssl_connection = 0;
-			i.ietf_version_or_minus_one = -1;
-			i.client_exts = exts;
-			i.protocol = protocols[0].name;
-			i.method = NULL;
-			wsi = lws_client_connect_via_info(&i);
-			if(wsi == NULL) {
-				JANUS_LOG(LOG_WARN, "Error attempting reconnection...\n");
-				continue;
+			janus_wsevh_connect_attempt(NULL);
+			if (!wsi) {
```

Comment:
Code style: no space between 'if` and the bracket.

Reply:
ack